src/fielddef.ts

```diff
@@ -156,6 +156,8 @@ export function field(fieldDef: FieldDef, opt: FieldRefOption = {}) {
 
   if (opt.datum) {
     field = `datum["${field}"]`;
+  } else if (field !== undefined && field.indexOf('.') !== -1) { // dot in fieldname
```

Comment:
I'll just mention here that in R, `.` is almost always referring to a dot within the column name, as opposed to nesting. Even the basic example datasets like `iris` include columns such as `sepal.length`. I agree if we always wrap in brackets then nested fields are impossible, but it will mean that potential libraries wrapping Vega-Lite in R may need to always wrap fields in brackets internally to be usable, which is ok but should be documented well.